js/ui/widget/ui.search_box_mixin.js

```diff
@@ -68,6 +69,21 @@ module.exports = {
         });
     },
 
+    _initDataSource: function() {
+        this.callBase();
+
+        if(this._applySourceOptions()) {
```

Comment:
dataSource instance will be created in callBase. It is not need to check this here. Check `searchEnabled` option only